api/src/middleware/team.js

```diff
@@ -23,6 +24,45 @@ async function validateTeamId(req, res, next) {
   next();
 }
 
+async function validateTeamRepo(req, res, next) {
+  const { ghToken } = req.body;
+  const { data } = await axios.get('https://api.github.com/user',
+    {
+      headers: { Authorization: `token ${ghToken}` },
+    });
+  if (data.errors) {
+    next(
+      fillErrorObject(400, 'Validation error: user doesnt exist!', [data.errors[0].detail]),
+    );
+  }
+
+  // Creating repoName
+  const ghUsername = data.login;
+  const repoName = `${ghUsername}.github.io`;
+  try {
+    const repoValidator = await axios.get(`https://api.github.com/repos/${ghUsername}/${repoName}`, {
+      headers: {
+        Authorization: `token ${ghToken}`,
+        Accept: 'application/vnd.github.v3+json',
+      },
+    });
+    if (repoValidator.status !== 200) {
+      next(
+        fillErrorObject(404, 'GH pages not found!', [
+          'GitHub Repo doesnt exist for this team!',
+        ]),
+      );
+    }
+    next();
+  } catch (error) {
+    next(
+      fillErrorObject(404, 'GH pages not found!', [
+        'GitHub Repo doesnt exist for this team!',
+      ]),
+    );
+  }
```

Comment:
What's the difference between these two? Shouldn't the next() in the catch block be a different error like server error?

Reply:
Good catch!